easybuild/framework/easyconfig/types.py

```diff
@@ -112,7 +112,7 @@ def convert_value_type(val, typ):
     return res
 
 
-def to_toolchain(tcspec):
+def to_nameversion_dict(tcspec, convert_to='dict'):
```

Comment:
also, depending on `convert_to`, this may yield a `tuple` rather than a `dict`, so rename this to `to_name_version`?